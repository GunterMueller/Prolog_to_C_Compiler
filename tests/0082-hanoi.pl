%	towers of hanoi ( hanoi ) for 8 disks

hanoi :- hanoi(8).

hanoi(N) :- move(N,left,center,right).

move(0,_,_,_) :- !.
move(N,A,B,C) :- M is N-1, move(M,A,C,B), inform(A,B), move(M,C,B,A).

inform(A,B) :- write([move,disk,from,A,to,B]), nl, fail.
inform(_,_).

:- initialization hanoi.


