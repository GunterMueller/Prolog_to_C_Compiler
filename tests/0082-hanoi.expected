[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,center,to,left]
[move,disk,from,center,to,right]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
[move,disk,from,right,to,left]
[move,disk,from,center,to,left]
[move,disk,from,right,to,center]
[move,disk,from,left,to,right]
[move,disk,from,left,to,center]
[move,disk,from,right,to,center]
